### PR TITLE
View columns should be able to be referenced with the schema name prefixed

### PR DESCRIPTION
**Pull request by stewartpark**
Fixes https://github.com/prestodb/presto/issues/9173

The base of the code fix is from the above mentioned issue. We were having the same problem, so I thought I'd start the work with the code given in the issue.

Currently, Metabase(a popular dashboard/visualization tool) references its columns with all parts(schema, table/view name, column name) and it breaks only on Presto views.

Considering the fact that a simple query that works with a table doesn't work with a view, I think it should be addressed!

### CODE REVIEW

**Comment by rmgpinto** [comment]
Can this PR be merged, please?

**Comment by rschlussel** [comment]
Hi @stewartpark I can review this today or tomorrow.  In the meantime can you squash the two commits together so the testcase is in the same commit as the change?  Also, I see the Travis build failed due to a planning failure in TestTpcdsCostBasedPlan.  Can you look at why that failed and fix?  It looks like some failure planning a "with" clause.

**Comment by findepi** [comment]
> Can this PR be merged, please? @martint

@stewartpark @rmgpinto You may want to open a PR @ https://github.com/prestosql/presto

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/Field.java [hunk 1]
```diff
@@ -122,7 +123,27 @@ public boolean isAliased()
...
+            return false;
+        }
+
+        int idx = 0;
+        boolean match = true;
+
+        QualifiedObjectName origin = originTable.get();
+        List<String> prefixes = prefix.get().getParts();
+
+        switch (prefixes.size()) {
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/Field.java]: The implicit loop here is unexpected for a switch statement,  which makes it hard to follow and error prone.

Can you change it to a series of if statements instead?
e.g. something like
```
if (prefixes.size = 0) {
    return false;
}

if(prefixes.size() = 3 && !origin.getCatalogName()...) {
     return false;
}
if (prefixes.size() >= 2 && origin.getSchemaName()...) {
     return false;
}
return origin.getObjectName()...;
```

Comment (stewartpark) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/Field.java]: Good point! Revised the code to be more explicit! Please see if this looks okay!

**Comment by stewartpark** [comment]
Thanks for the thorough review, @rschlussel! The failing test seems to pass locally now so if you could take another look at this one! :pray: 

@findepi, Will also do that as well!

**Comment by rschlussel** [comment]
@stewartpark The code changes look good, but the test is still failing and it also fails when I run your code locally. Can you please take a look?

A good way to test it out:
1. Start the TpcdsQueryRunner (in debug mode if you'll want to add breakpoints)-- this will start basically a local instance of presto with just the tpcds catalog running at localhost:8080 
2. Start up the presto cli locally and use catalog/schema of tpcds.sf1
3. then run the query in the cli

**Comment by rmgpinto** [comment]
@stewartpark do you have time to continue fixing this issue?

**Comment by stewartpark** [comment]
@rmgpinto I'll continue working on this this weekend and request another review from the original reviewer. :pray:

**Comment by findepi** [comment]
@stewartpark @rmgpinto 
The bug is already fixed in prestosql. You can grab a build from https://prestosql.io/download.html and give it a try.

**Comment by rmgpinto** [comment]
@findepi do you know which version fixed the bug?

**Comment by rmgpinto** [comment]
I'm running AWS EMR so no prestosql there.

**Comment by electrum** [comment]
@rmgpinto This was fixed in release 304: https://prestosql.io/docs/current/release/release-304.html

> Fix failure when a column is referenced using its fully qualified form.

**Comment by raghavsethi** [comment]
@rschlussel Could you merge this?

**Comment by rschlussel** [comment]
Here's a PR with the backport for the fix from prestosql - https://github.com/prestodb/presto/pull/12565

**Comment by stewartpark** [comment]
Ah, glad to hear it was already fixed. Closing in favor of the backport PR. Sorry for my late follow up
